Secret-agency-web/src/main/react/secret-agency/src/App.tsx

```diff
@@ -110,6 +111,10 @@ class App extends React.Component<{}, IState> {
                     <button className="logout-button" type={'button'} onClick={this.onLogout}>Log out</button>
                     <Route path={`${ROUTING_URL_BASE}/agents`} component={AgentsPage}/>
                     <Route path={`${ROUTING_URL_BASE}/departments`} component={DepartmentsPage}/>
+                    <Route path={`${ROUTING_URL_BASE}/reports`} component={() => (
+                        <ReportsPage authenticatedUserId={this.state.authenticatedAgent!.id}
```

Comment:
jeste toto by se dalo pekne vyresit v `ability.ts`
userovi by se mohlo pridat neco jako 
`can(['read', 'update'], ['Report'], { authorId: agent.id })`
a adminovi:
`can(['read', 'update'], ['Report'])`

nebo whatever jim chces nadeifnovat, aby mohli delat, pak bys sis toto podle me vubec nemusel predavat

Reply:
Mohl bys to udělat, prosím? Já jsem se pustil do té dokumentace REST rozhraní.  :)

Reply:
Nebo to udělám až zítra.

Reply:
ok, akorat si z toho musim udelat novou vetev, protoze to mas na svojem repu, tak vytvorim novy PR.

jaka ma byt business logika? agentovi zobrazovat jen jeho, adminovi vsechny? edit se zatim neresi ze

Reply:
Jo, zatím šlo jen o to, aby se adminovi zobrazovali všechny, uživateli pouze jeho a neměl v tabulce sloupec agent, který běžně určuje, kdo daný report napsal.

Reply:
Ale nevím, jestli ti v tomto případě moc to ability.ts pomůže, protože id tam předávám z důvodu filtrování řádků a sloupců, když přihlášený uživatel není admin.

Reply:
v tom ability muzes nadefinovat pravidla vazana primo na nejake ID, takze myslim ze ot bude cajk, kazdopadne zkusim a uvidim 😄

Reply:
Nemůžeš to mergnout a upravit potom? :)
Abych si zachoval autorství. 😄

Reply:
autorstvi radku kodu zustane stejne i tak 😉

Reply:
A commity asi taky, ne?

Reply:
jj proces je stejny jako kdybych se checkoutnul ze nejake na tomto repu

viz. https://help.github.com/articles/checking-out-pull-requests-locally/

Reply:
Ok :)

Reply:
tak po dlouhem badani jsem zjistil, ze to nefunguje na interfacech 🤦‍♂️  takze bychom z toho IReport (a dalsich, ktere bychom chteli checkovat) museli udelat classu.

furt by nam to mohlo usetrit par ifu a radku kodu, co myslis?

Reply:
Já jsem se na to nedíval, takže nevím, jak to funguje. Zatím bych tu knihovnu použil jen na místech, kde funguje v pořádku. Interface bych na classu nepřepisoval.